### PR TITLE
#83 Added secret file provider

### PR DESCRIPTION
**Pull request by nazar256**
## Related Issues
#83

## Description
Added secret file provider. It maps secret is presented as file containing a secret string (i.e. docker secret mounted into container) into environment variable. For instance, for applications which do not support secret files themselves.

## Motivation and Context
Mounting secrets into docker container is considered as best practice opposed to passing them as environment variables. However only few applications support secret files out of the box.
https://github.com/SpectralOps/teller/issues/83

## How Has This Been Tested?
Tested it manually. For instance with the default config `./teller run env`. Covered with unit tests in similar way as other providers.
Tested in linux_amd64 environment manually and with `make test`.
No possible side effects (on other code) are expected.

# Checklist
- [x] Tests
- [x] Documentation
- [x] Linting

### CODE REVIEW

File: pkg/providers/secretfile.go [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+			return false, nil
+		}
+
+		return false, err
+	}
+
+	return true, nil
+}
+
+func (d *SecretFileReader) Delete(p string) error {
```
Comment (kaplanelad) [inline pkg/providers/secretfile.go:70]: This function in used?

File: README.md [hunk 2]
```diff
@@ -736,6 +736,33 @@ dotenv:
...
+* Key format 
+  * `env` - env key like
+
+
+### Example Config
+
+You can specify secrets mount directory (usually `/run/secrets`) or map individual secret to env variable name.
+
+```yaml
+dotenv:
```
Comment (kaplanelad) [inline README.md:759]:
```suggestion
secretfile:
```

**Review by kaplanelad — CHANGES_REQUESTED** [review]
Hey @nazar256, thanks for adding this provider. 
Can you please check my litter comments?

**Comment by nazar256** [comment]
Already solved by `filesystem` plugin.

**Pull request closed without merging**